mailet/icalendar/src/test/java/org/apache/james/transport/mailets/ICALToHeadersTest.java

```diff
@@ -0,0 +1,187 @@
+            .build();
+
+        testee.service(mail);
+
+        assertThat(mail.getMessage().getHeader(ICALToHeader.X_MEETING_UID)).isNull();
+    }
+
+    @Test
+    public void serviceShouldWriteSingleICalendarToHeaders() throws Exception {
+        Calendar calendar = new CalendarBuilder().build(ClassLoader.getSystemResourceAsStream("ics/meeting.ics"));
+        ImmutableMap<String, Calendar> icals = ImmutableMap.<String, Calendar>builder()
+            .put("key", calendar)
+            .build();
+
+        testee.init(FakeMailetConfig.builder().build());
+        Mail mail = FakeMail.builder()
+            .mimeMessage(MimeMessageBuilder.defaultMimeMessage())
+            .attribute(ICALToHeader.ATTRIBUTE_DEFAULT_NAME, icals)
+            .build();
+
+        testee.service(mail);
+
+        assertThat(mail.getMessage().getHeader(ICALToHeader.X_MEETING_METHOD)).containsOnly("REQUEST");
+        assertThat(mail.getMessage().getHeader(ICALToHeader.X_MEETING_UID))
+            .containsOnly("f1514f44bf39311568d640727cff54e819573448d09d2e5677987ff29caa01a9e047feb2aab16e43439a608f28671ab7c10e754ce92be513f8e04ae9ff15e65a9819cf285a6962bc");
+        assertThat(mail.getMessage().getHeader(ICALToHeader.X_MEETING_DTSTAMP)).containsOnly("20170106T115036Z");
+        assertThat(mail.getMessage().getHeader(ICALToHeader.X_MEETING_RECURRENCE_ID)).isNull();
+        assertThat(mail.getMessage().getHeader(ICALToHeader.X_MEETING_SEQUENCE)).containsOnly("0");
+    }
+
+    @Test
+    public void serviceShouldWriteOnlyOneICalendarToHeaders() throws Exception {
+        Calendar calendar = new CalendarBuilder().build(ClassLoader.getSystemResourceAsStream("ics/meeting.ics"));
+        Calendar calendar_2 = new CalendarBuilder().build(ClassLoader.getSystemResourceAsStream("ics/meeting.ics"));
+        ImmutableMap<String, Calendar> icals = ImmutableMap.<String, Calendar>builder()
+            .put("key", calendar)
+            .put("key2", calendar_2)
+            .build();
+
+        testee.init(FakeMailetConfig.builder().build());
+        Mail mail = FakeMail.builder()
+            .mimeMessage(MimeMessageBuilder.defaultMimeMessage())
+            .attribute(ICALToHeader.ATTRIBUTE_DEFAULT_NAME, icals)
+            .build();
+
+        testee.service(mail);
+
+        assertThat(mail.getMessage().getHeader(ICALToHeader.X_MEETING_UID)).hasSize(1);
+    }
+
+    @Test
+    public void serviceShouldNotFailOnEmptyMaps() throws Exception {
+        ImmutableMap<String, Calendar> icals = ImmutableMap.<String, Calendar>builder()
+            .build();
+
+        testee.init(FakeMailetConfig.builder().build());
+        Mail mail = FakeMail.builder()
+            .mimeMessage(MimeMessageBuilder.defaultMimeMessage())
+            .attribute(ICALToHeader.ATTRIBUTE_DEFAULT_NAME, icals)
+            .build();
+
+        testee.service(mail);
+
+        assertThat(mail.getMessage().getHeader(ICALToHeader.X_MEETING_UID)).isNull();
+    }
+}
```

Comment:
some ics doesn't contain all the data you expect, and then you'll not add these headers. Please add a test showing this